cuML/CMakeLists.txt

```diff
@@ -79,27 +89,60 @@ list(GET GPU_ARCHS -1 ptx)
 set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode arch=compute_${ptx},code=compute_${ptx}")
 ## end nvcc options
 
+###################################################################################################
+# - include paths ---------------------------------------------------------------------------------
+
 include_directories(src
-  ${GTEST_DIR}/googletest/include
-  ${MLPRIMS_DIR}/src
-  ${MLPRIMS_DIR}/src/dbscan
-  ${MLPRIMS_DIR}/test
-  ${CUTLASS_DIR}
-  ${CUTLASS_DIR_2}
-  ${CUB_DIR})
+                    ${GTEST_DIR}/googletest/include
+                    ${MLPRIMS_DIR}/src
+                    ${MLPRIMS_DIR}/src/dbscan
+                    ${MLPRIMS_DIR}/test
+                    ${CUTLASS_DIR}
+                    ${CUTLASS_DIR_2}
+                    ${CUB_DIR})
 
 add_subdirectory(${GTEST_DIR}/googletest ${PROJECT_BINARY_DIR}/googletest)
 
 # Append source file in recursive manner, append header files to target for work with them in IDE
 file(GLOB_RECURSE ml_prims_header "${MLPRIMS_DIR}/src/*.h" "${MLPRIMS_DIR}/src/*.hpp")
 file(GLOB_RECURSE cuml_test_cuda_sources "test/*.cu")
 
+###################################################################################################
+# - build libcuml shared library ------------------------------------------------------------------
+
+add_library(cuml SHARED
```

Comment:
Due to possibility of having this kernels cuml after this change will not be header only and will be distributed as shared library? (I.e. we shifting from declared principle in this README https://github.com/rapidsai/cuml/tree/master/cuML `cuML is implemented as header only C++/CUDA libraries`)

Reply:
The main reason for this change actually stems from wanting to fully separate the Cython bindings building from compilation of the library, otherwise I didn't find a way to avoid Cython needing to compile the CUDA files. But this is for Python only, so we could make this library required for the cuML python package, but not for pure C++ development purposes.

Reply:
I'm not familiar with Cython binding. Another comments are fine with me. Thanks for your work.

Reply:
This looks good. I think we can merge it.